0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,62 +1,127 @@
+9. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+10. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+#### Invoice
+
+1. The sender's SPSP client queries the receiver endpoint.
+2. The receiver endpoint responds with the receiver info, including the invoice status, amount, and currency code.
+3. The sender's SPSP client submits the sender's info, including the sender's ILP address to the receiver endpoint.
+4. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+5. The sender's SPSP client uses its ILP module to get a quote in their currency or asset type for the ILP transfer.
+6. The sender accepts the quote.
+7. The sender's SPSP client uses its ILP module to initiate the ILP transfer.
+8. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+9. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
 
-### Query
+## Specification
 
-Any SPSP recipient will create a receiving endpoint called the *receiver*. The sender queries this endpoint to get information about the type of payment that can be made to this receiver:
+The receiver endpoint is a URI used by the sender to query information about the sender and set up payments. The receiver URI MAY contain query string parameters. The sender SHOULD treat the URI as opaque.
 
+The receiver endpoint MUST respond to HTTP `GET` and `POST` requests in the following manner:
+
+### Query (`GET <receiver>`)
+
+The sender queries the receiver endpoint to get information about the type of payment that can be made to this receiver:
+
+#### Request
 ``` http
 GET /api/receivers/bob HTTP/1.1
 Host: red.ilpdemo.org
 Accept: application/json
 ```
+
+#### Response
 ``` http
 HTTP/1.1 200 OK
 Content-Type: application/json
 
 {
   "type": "payee",
-  "ledger": "https://red.ilpdemo.org/ledger",
-  "account": "https://red.ilpdemo.org/ledger/accounts/bob",
-  "payments": "https://red.ilpdemo.org/api/receiver/bob/payments"
+  "account": "ilpdemo.red.bob",
+  "currency_code": "USD",
```

Comment:
Eventually, we'll probably want to overhaul the asset system. I'm thinking that there would be a very generic, catch-all way of describing an asset uniquely and then a bunch of templates that allow you to provide richer definitions for certain asset types, e.g. currencies, bonds, etc.

I specifically **don't** want us to get distracted by that right now, since it's not part of the ILP layer and it's quite a large task to do properly.